### PR TITLE
OCP feature gate

### PR DESCRIPTION
**Pull request by lihongyan1**
OCP MODE

Add step to enable OCP mode
Update pre-merge job to enable OCP mode after install ObO
Update test step to pass the operator installation namespace

### CODE REVIEW

**Comment by lihongyan1** [comment]
/pj-rehearse periodic-ci-rhobs-observability-operator-main-ocp-4.15-coo-stage

**Comment by lihongyan1** [comment]
/pj-rehearse periodic-ci-rhobs-observability-operator-main-ocp-4.15-coo-stage

**Comment by lihongyan1** [comment]
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Comment by lihongyan1** [comment]
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Comment by lihongyan1** [comment]
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Comment by lihongyan1** [comment]
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Comment by lihongyan1** [comment]
/pj-rehearse periodic-ci-rhobs-observability-operator-main-ocp-4.15-coo-stage

**Comment by lihongyan1** [comment]
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Comment by lihongyan1** [comment]
/pj-rehearse periodic-ci-rhobs-observability-operator-main-ocp-4.15-coo-stage

**Comment by lihongyan1** [comment]
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Comment by lihongyan1** [comment]
Prow job failure is expected for https://github.com/rhobs/observability-operator/pull/531 is not merged, I have built test-harness image with the PR and test with the image, everything works well
```
$ podman run --user 0 -it -v $KUBECONFIG:/tmp/kubeconfig:Z --env 'KUBECONFIG=/tmp/kubeconfig' --entrypoint /bin/bash quay.io/lihongyan1/observability-operator-test-harness:0.3.0-test-1
bash-4.2# ls
e2e.test  go-junit-report  osd-e2e.sh
bash-4.2# ./e2e.test -test.v -operatorInstallNS=coo
=== RUN   TestMonitoringStackController
=== RUN   TestMonitoringStackController/Defaults_are_applied_to_Monitoring_CR
=== RUN   TestMonitoringStackController/Defaults_are_applied_to_Monitoring_CR/empty-stack
```

**Comment by lihongyan1** [comment]
@jan--f Please help review

**Comment by lihongyan1** [comment]
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Comment by lihongyan1** [comment]
@simonpasquier Please help review and merge

File: ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main.yaml [hunk 1]
```diff
@@ -60,9 +60,21 @@ tests:
...
+        tries=30
+        while [[ $tries -gt 0 ]] && ! oc -n coo rollout status deploy/obo-prometheus-operator; do
+          sleep 10
+          ((tries--))
+        done
+        oc wait -n coo --for=condition=Available deploy/obo-prometheus-operator --timeout=300s
+        oc wait -n coo --for=condition=Available deploy/obo-prometheus-operator-admission-webhook --timeout=300s
+        oc wait -n coo --for=condition=Available deploy/observability-operator --timeout=300s
+        CSV_NAME=$(oc -n coo get csv | grep 'observability-operator' | awk '{print $1}')
+        oc -n coo get csv "$CSV_NAME" -oyaml > "${SHARED_DIR}/coo-csv.yml"
```
Comment (jan--f) [inline ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main.yaml:77]: I have no doubts this works.
Have you explored editing the CSV before the bundle is build? I don't know the details but I assume there is a build step where we build the bundle from the merge PR. If we edit the CSV to add our cli arg there, we don't have to wait for the operator to become available and then change it. We could also add a `kustomize` step on obo specifically for this.

Comment (lihongyan1) [reply ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main.yaml:77]: From ci-operator doc, we can see that images can be substituted when build the bundle andhave no way to edit the csv for now, also checked with ci-operator team for the issue https://redhat-internal.slack.com/archives/CBN38N3MW/p1727230485893919

Comment (jan--f) [reply ci-operator/config/rhobs/observability-operator/rhobs-observability-operator-main.yaml:77]: Thanks for checking on this!

**Comment by jan--f** [comment]
/lgtm
/pj-rehearse pull-ci-rhobs-observability-operator-main-observability-operator-e2e

**Pull request merged**